src/zkevm_specs/evm/typing.py

```diff
@@ -666,19 +667,42 @@ def _append(
         return self
 
 
+class KeccakCircuit:
+    rows: List[KeccakTableRow]
+
+    def __init__(self) -> None:
+        self.rows = []
+
+    def add(self, data: bytes, r: FQ) -> KeccakCircuit:
+        output = RLC(keccak256(data), r, n_bytes=32)
+        acc_input = FQ.zero()
+        for i in range(len(data)):
+            acc_input = acc_input * r + data[i]
```

Comment:
Maybe use standard API so that we don't worry about the order of bytes?
```suggestion
        acc_input = RLC(data, r, n_bytes=len(data))
```